### ISSUE TITLE
Softmax poor performance

### ISSUE THREAD

**Issue by emartinezs44** [intel/analytics-zoo#21]
The **com.intel.analytics.zoo.pipeline.api.keras.layers.SoftMax** layer shows a poor performance in comparison with other frameworks using the same hardware. In essence the operation from TensorNumeric _**vexp**_ . The tests that I have done with an input Tensor[Float] with dims as Array(1, 12, 512, 512) and Storage of 3145728 length it takes a mean of 0.65 seg per inference. As a result, models with attention that use this function in each layer to calculate the attention vector take so long. The Softmax layer in Torch is much faster in the same hardware. Any idea about this? how much faster is Mkl-dnn?
